rotkehlchen/chain/evm/decoding/rainbow/decoder.py

```diff
@@ -0,0 +1,336 @@
+
+from rotkehlchen.assets.asset import Asset, CryptoAsset, EvmToken
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value, asset_raw_value
+from rotkehlchen.chain.evm.decoding.constants import ERC20_OR_ERC721_TRANSFER
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    ActionItem,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.evm.decoding.weth.decoder import WETH_DEPOSIT_TOPIC, WETH_WITHDRAW_TOPIC
+from rotkehlchen.chain.evm.structures import EvmTxReceiptLog
+from rotkehlchen.constants.assets import A_ETH
+from rotkehlchen.errors.asset import UnsupportedAsset
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.types import ChecksumEvmAddress, EVMTxHash, EvmTransaction
+from rotkehlchen.utils.misc import bytes_to_address
+
+from .constants import CPT_RAINBOW_SWAPS
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+    from rotkehlchen.user_messages import MessagesAggregator
+
+
+class RainbowCommonDecoder(DecoderInterface):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            router_address: ChecksumEvmAddress,
+    ) -> None:
+        super().__init__(evm_inquirer, base_tools, msg_aggregator)
+        self.router_address = router_address
+
+    def _resolve_asset(self, asset: Asset) -> EvmToken | CryptoAsset:
+        """Resolve a asset into evmtoken or cryptoasset
+
+        Args:
+            asset (Asset): Asset to be resolve
+
+        Returns:
+            EvmToken | CryptoAsset: Asset resolved
+        """
+        return self.evm_inquirer.native_token if asset == A_ETH else asset.resolve_to_evm_token()
+
+    def _tansform_decoded_events(
+            self,
+            decoded_events: list['EvmEvent'],
+            sender: ChecksumEvmAddress
+    ) -> tuple[Optional['EvmEvent'], Optional['EvmEvent']]:
+        """Transform the receive and spend events to rainbow events
+
+        Returns:
+            in_event (EvmEvent | None): input event, the receive the tokens
+            out_event (EvmEvent | None): ouput event, the spend the tokens
+        """
+        in_event = out_event = None
+        for event in decoded_events:
+            if (event.event_type == HistoryEventType.INFORMATIONAL and event.event_subtype == HistoryEventSubType.APPROVE):
+                # Remove the approve from rainbow router to aggretor
+                # to avoid show unnecessary events 
+                decoded_events.remove(event)
```

Comment:
don't do this. We always leave them as they are useful to identify infinite approvals and perform other analytics